content/6_Advanced/Parallel_Binary_Search.mdx

```diff
@@ -0,0 +1,141 @@
+---
+id: parallel-binary-search
+title: 'Parallel Binary Search'
+author: Muaath Alqarni
+description: 'Offline way to do multiple binary search on answer efficiently'
+prerequisites:
+  - binary-search
+  - divide-and-conquer
+frequency: 0
+---
+
+
+## Resources
+
+<Resources>
+	<Resource source="CF" title="Parallel Binary Seach [tutorial]" url="45578" />
+	<Resource source="SPBook" title="Parallel Binary Seach" url="https://kostka.dev/sp/spbook.pdf#page=22" />
+</Resources>
+
+<FocusProblem problem="sam" />
+
+## Solution - New Roads Queries
+Let's think about how to use binary to answer single query.
+We can see that connvectivity of the nodes is a monotonic function.
+
+Now let's use parallel binary search:
+
+First, sort the queries by median.
+
+Then sweep through the array of edges and union them in the DSU,
+whenevery there is a query with median = $i$, if the two nodes where connected in DSU, it means the answer is $\leq i$, otherwise it means $\gt i$.
```

Comment:
```suggestion
Whenever there is a query with median = $i$, if the two nodes where connected in DSU, it means the answer is $\leq i$, otherwise it means $\gt i$.
```